apps/arweave/src/ar_util.erl

```diff
@@ -101,15 +101,28 @@ invert_map(Map) ->
     ).
 
 
-%% @doc Parse a string representing a remote host into our internal format.
+%%--------------------------------------------------------------------
+%% @doc Parse a string representing a remote host into our internal
+%%      format.
+%% @end
+%%--------------------------------------------------------------------
+-spec parse_peer(Hostname) -> Return when
```

Comment:
it looks like `parse_peer(String)` always returns a list (sometimes a 1 element list), but `parse_peer{IP, PORT})` returns a single element. How hard would it be to have all `parse_peer` matches return a list? A little worried about potential complexity in the future re: return type

Reply:
e.g. here: https://github.com/ArweaveTeam/arweave/pull/720/files#diff-de37132d068e49cb11eb29b5c880b8ef2454d93f79686171b6cff9392824a2c5R349

Reply:
oh I thought `parse_peer/1` sometimes returns a tuple and not a list? Here: https://github.com/ArweaveTeam/arweave/pull/720/files#diff-1f26621f4ea9ef47e071afce0f4ce9fc3e6bacc76089ac9037c00e0c07a5ce9eR129

Reply:
i.e.
```
parse_peer({IP, Port}) ->
	{A, B, C, D} = parse_peer(IP),
	{A, B, C, D, parse_port(Port)}.
```

It's this match that I think will cause a problem, since it means `parse_peer/1` sometimes returns a list and sometimes a returns a tuple.